flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGateTest.java

```diff
@@ -126,6 +126,41 @@ public void testBasicGetNextLogic() throws Exception {
 		}
 	}
 
+	/**
+	 * Tests that the compressed buffer will be decompressed after calling {@link SingleInputGate#getNext()}.
+	 */
+	@Test
+	public void testGetCompressedBuffer() throws Exception {
+		int bufferSize = 1024;
+		String compressCodec = "LZ4";
+		BufferCompressor compressor = new BufferCompressor(bufferSize, compressCodec);
+		BufferDecompressor decompressor = new BufferDecompressor(bufferSize, compressCodec);
+
+		SingleInputGate inputGate = new SingleInputGateBuilder().setBufferDecompressor(decompressor).build();
```

Comment:
It is better to close gate afterwards via `try (SingleInputGate inputGate = new SingleInputGateBuilder().setBufferDecompressor(decompressor).build())` to avoid potential resource leak.